diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/classes/InlineClassesRule.kt

```diff
@@ -0,0 +1,50 @@
+package org.cqfn.diktat.ruleset.rules.classes
+
+import org.cqfn.diktat.common.config.rules.RulesConfig
+import org.cqfn.diktat.ruleset.constants.EmitType
+import org.cqfn.diktat.ruleset.constants.Warnings.INLINE_CLASS_CAN_BE_USED
+import org.cqfn.diktat.ruleset.utils.getFirstChildWithType
+
+import com.pinterest.ktlint.core.Rule
+import com.pinterest.ktlint.core.ast.ElementType.CLASS
+import com.pinterest.ktlint.core.ast.ElementType.FINAL_KEYWORD
+import com.pinterest.ktlint.core.ast.ElementType.INTERNAL_KEYWORD
+import com.pinterest.ktlint.core.ast.ElementType.MODIFIER_LIST
+import com.pinterest.ktlint.core.ast.ElementType.PRIVATE_KEYWORD
+import com.pinterest.ktlint.core.ast.ElementType.PROTECTED_KEYWORD
+import com.pinterest.ktlint.core.ast.ElementType.PUBLIC_KEYWORD
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.psi.KtClass
+
+class InlineClassesRule(private val configRule: List<RulesConfig>) : Rule("inline-classes") {
+    private var isFixMode: Boolean = false
+    private lateinit var emitWarn: EmitType
+
+    override fun visit(
+            node: ASTNode,
+            autoCorrect: Boolean,
+            emit: EmitType
+    ) {
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        if (node.elementType == CLASS) {
+            handleClasses(node)
+        }
+    }
+
+    private fun handleClasses(node: ASTNode) {
+        if ((node.psi as KtClass).getProperties().size == 1 && !(node.psi as KtClass).hasExplicitPrimaryConstructor()) {
+            val modList = node.getFirstChildWithType(MODIFIER_LIST)
```

Comment:
Here you can also use `node.psi as KtClass` to get all modifiers and pass to this method node as psi

Reply:
Fixed